### PR TITLE
Web share target support

### PR DESCRIPTION
**Pull request by chenlevy24**

### CODE REVIEW

**Comment by chenlevy24** [comment]
Extended the webManifest interface to support web share target completely. Moreover, added `escapeJsonString` method for formatting the shareTarget JSON on `build.gradle`.

File: packages/core/template_project/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -55,14 +55,27 @@
             android:name="twa_generator"
             android:value="@string/generatorApp" />
 
+        <meta-data
+            android:name="android.support.customtabs.trusted.METADATA_SHARE_TARGET"
+            android:resource="@string/shareTarget"/>
+
```
Comment (andreban) [inline packages/core/template_project/app/src/main/AndroidManifest.xml]: We only want to add this when Share Target is enabled:
```suggestion
 <% if (shareTarget) { %>
        <meta-data
            android:name="android.support.customtabs.trusted.METADATA_SHARE_TARGET"
            android:resource="@string/shareTarget"/>
<% } %>
```

File: packages/core/template_project/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -55,14 +55,27 @@
...
-        
+
+        <intent-filter>
+            <action android:name="android.intent.action.SEND" />
+            <action android:name="android.intent.action.SEND_MULTIPLE" />
+            <category android:name="android.intent.category.DEFAULT" />
+            <data android:mimeType="audio/*" />
+            <data android:mimeType="image/*" />
+            <data android:mimeType="video/*" />
+        </intent-filter>
```
Comment (andreban) [inline packages/core/template_project/app/src/main/AndroidManifest.xml]: Same for the `intent-filter`:

```suggestion
 <% if (shareTarget) { %>
        <intent-filter>
            <action android:name="android.intent.action.SEND" />
            <action android:name="android.intent.action.SEND_MULTIPLE" />
            <category android:name="android.intent.category.DEFAULT" />
            <data android:mimeType="audio/*" />
            <data android:mimeType="image/*" />
            <data android:mimeType="video/*" />
        </intent-filter>
<% } %>
```

**Review by andreban — CHANGES_REQUESTED** [review]
A couple of changes around only including the meta-tag and intent-filter when shareTarget is enabled.

File: packages/core/template_project/app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -105,11 +105,28 @@
             <meta-data android:name="android.support.customtabs.trusted.FALLBACK_STRATEGY"
                 android:value="@string/fallbackType" />
 
+            <% if (shareTarget) { %>
```
Comment (PEConn) [inline packages/core/template_project/app/src/main/AndroidManifest.xml:108]: Why not combine this section and the one you added below? Put both bits inside a single `if (shareTarget)`?

File: packages/core/src/lib/util.ts [hunk 4]
```diff
@@ -276,3 +276,16 @@ export async function getWebManifest(webManifestUrl: URL): Promise<WebManifestJs
   }
   return await response.json();
 }
+
+/**
+ * Given a string of a JSON, the function retrns an escaped string representing that string.
```
Comment (PEConn) [inline packages/core/src/lib/util.ts:281]: nit: retrns -> returns

File: packages/core/src/lib/util.ts [hunk 5]
```diff
@@ -276,3 +276,16 @@ export async function getWebManifest(webManifestUrl: URL): Promise<WebManifestJs
...
+ * eg: Turns every " instance into \\".
+ *
+ * @param {string} stringToReplace the string before the manipulation.
+ * @returns {string} the string after the manipulation.
+ */
+export function escapeJsonString(stringToReplace: string): string {
+  // The 'g' flag is for replacing all of the instances.
+  const regExp = new RegExp('\"', 'g');
+  return stringToReplace.replace(regExp, '\\\\\"');
```
Comment (PEConn) [inline packages/core/src/lib/util.ts:290]: I am confused what this bit does - the `\\\\\"` specifically?

Comment (andreban) [reply packages/core/src/lib/util.ts:290]: It replaces `"` with `\\"`. We end up with quite a few escapes for that.

Comment (chenlevy24) [reply packages/core/src/lib/util.ts:290]: I figured that will be confusing so I mentioned it in the function's documentation.

**Review by andreban — APPROVED** [review]
LGTM

**Review by PEConn — APPROVED** [review]

**Pull request merged**